src/org/ods/services/OpenShiftService.groovy

```diff
@@ -171,4 +173,69 @@ class OpenShiftService {
       label: "Get phase of build ${buildId}"
     ).trim().toLowerCase()
   }
+  
+  public String getContainerForImage (String projectId, String rc, String image) {
+    script.sh(
+      script: """oc -n ${projectId} get rc ${rc} -o jsonpath='{.spec.template.spec.containers[?(contains .image "${image}")].name}'""",
+    )
+  }
+  
+  // Gets pod of deployment
+  Map getPodDataForDeployment(String componentId, String rc) {
+    
+    if (!automaticImageChangeTriggerEnabled(componentId)) {
+      script.echo ("No change trigger found for ${componentId} - hence no pod created!")
+      return [ : ]
+    }
+    
+    def index = 5
+    while (index > 0)
+    {
+      def podStatus = script.sh(
+        script: "oc -n ${project} get pod -l deployment=${rc} -o jsonpath='{.items[*].status.phase}'",
+        returnStdout: true,
+        label: "Getting OpenShift pod data for deployment ${rc}"
+      )
+      if (podStatus && podStatus == "Running") {
+        break
+      } else {
+        script.sleep(5000)
+        index--
+      }
+    }
```

Comment:
This has some overlap with `getRolloutStatus`. I think we should settle on one way to check if a deployment is successful or not. At the moment, I'm not sure which way is better though :)

Reply:
@michaelsauter  - rollout != pod ... so I think both have their place?!

Reply:
Sure - I was just thinking if we should unify checking the phase. I think the code above is already true when the pod is started, but checking for the rollout is probably only complete when the healthcheck succeeds. Maybe we actually want the latter?